lib/_http_server.js

```diff
@@ -477,20 +478,25 @@ function connectionListener(socket) {
       // if the user never called req.read(), and didn't pipe() or
       // .resume() or .on('data'), then we call req._dump() so that the
       // bytes will be pulled off the wire.
-      if (!req._consuming && !req._readableState.resumeScheduled)
+      if (!req._consuming && !req._readableState.resumeScheduled) {
         req._dump();
+      }
 
       res.detachSocket(socket);
+      res._last = outgoing.length === 0;
```

Comment:
Again, it should be a separate clause for this thing. `_last` indicates that there are not going to be any requests on the TCP socket. While there definitely could be in case of `outgoing.length === 0`.